README.md
# test-m1-skylabs
TEST - M1

PUNTO 1:

Oltre a elencare qui di seguito le modifiche da apportare al codice, ho realizzato direttamente le modifiche che potrete visualizzare, utilizzando il file "menu.html".
- Per prima cosa ho inserito una griglia bootstrap, in modo da dare giusti spazi alle varie sezioni.
- Ho modificato il layout cambiando i colori di sfondo ed eliminato il colore dei bordi.
- Per le immagini del carrello, user e wishlist ho utilizzato delle icone Font Awesome, e inserito un margine per distanziarle.
- Ingrandito e dato un margine diverso al logo.
- Utilizzato un effetto dropdown per visualizzare il sottomenu di ogni link menu

********************************************************************************************************************************************************************

PUNTO 2

Lanciare file "quadrato.html".


****************************************************************************************************

PUNTO 3

Il progetto è composto dai segenti file:
 - function.php
 - homenews.php
 - news.php
 - singolnews.php
 - css/stylenews.css

Lanciare file homenews.php

****************************************************************************************************

PUNTO 4


Il progetto è composto dai segenti file:
 - prodotto.html
 - css/styleprodotto.css
 - cartella img

Lanciare file prodotto.html


news.txt
1;Titolo1;Mario Rossi;Questo è il corpo dell'articolo uno e sto scrivendo diverse cose di prova. Huic autem est illa dispar adiuncta ratio orationis quae alio quodam genere mentes iudicum permovet impellitque ut aut oderint aut diligant aut invideant aut salvum velint aut metuant aut sperent aut cupiant aut abhorreant aut laetentur aut maereant aut misereantur aut poenire velint aut ad eos motus deducantur si qui finitimi sunt et de propinquis ac talibus animi permotionibus. Atque illud optandum est oratori ut aliquam permotionem animorum sua sponte ipsi adferant ad causam iudices ad id quod utilitas oratoris feret accommodatam. Facilius est enim currentem ut aiunt incitare quam commovere languentem. Sin id aut non erit aut erit obscurius sicut medico diligenti priusquam conetur aegro adhibere medicinam non solum morbus eius cui mederi volet sed etiam consuetudo valentis et natura corporis cognoscenda est sic equidem cum adgredior in ancipiti causa et gravi ad animos iudicum pertractandos omni mente in ea cogitatione curaque versor ut odorer quam sagacissime possim quid sentiant quid existiment quid exspectent quid velint quo deduci oratione facillime posse videantur. 
2;Titolo2;Mario Verdi;Questo è il corpo dell'articolo due e sto scrivendo diverse cose di prova. Huic autem est illa dispar adiuncta ratio orationis quae alio quodam genere mentes iudicum permovet impellitque ut aut oderint aut diligant aut invideant aut salvum velint aut metuant aut sperent aut cupiant aut abhorreant aut laetentur aut maereant aut misereantur aut poenire velint aut ad eos motus deducantur si qui finitimi sunt et de propinquis ac talibus animi permotionibus. Atque illud optandum est oratori ut aliquam permotionem animorum sua sponte ipsi adferant ad causam iudices ad id quod utilitas oratoris feret accommodatam. Facilius est enim currentem ut aiunt incitare quam commovere languentem. Sin id aut non erit aut erit obscurius sicut medico diligenti priusquam conetur aegro adhibere medicinam non solum morbus eius cui mederi volet sed etiam consuetudo valentis et natura corporis cognoscenda est sic equidem cum adgredior in ancipiti causa et gravi ad animos iudicum pertractandos omni mente in ea cogitatione curaque versor ut odorer quam sagacissime possim quid sentiant quid existiment quid exspectent quid velint quo deduci oratione facillime posse videantur. 
3;Titolo3;Mario Bianchi;Questo è il corpo dell'articolo tre e sto scrivendo diverse cose di prova. Huic autem est illa dispar adiuncta ratio orationis quae alio quodam genere mentes iudicum permovet impellitque ut aut oderint aut diligant aut invideant aut salvum velint aut metuant aut sperent aut cupiant aut abhorreant aut laetentur aut maereant aut misereantur aut poenire velint aut ad eos motus deducantur si qui finitimi sunt et de propinquis ac talibus animi permotionibus. Atque illud optandum est oratori ut aliquam permotionem animorum sua sponte ipsi adferant ad causam iudices ad id quod utilitas oratoris feret accommodatam. Facilius est enim currentem ut aiunt incitare quam commovere languentem. Sin id aut non erit aut erit obscurius sicut medico diligenti priusquam conetur aegro adhibere medicinam non solum morbus eius cui mederi volet sed etiam consuetudo valentis et natura corporis cognoscenda est sic equidem cum adgredior in ancipiti causa et gravi ad animos iudicum pertractandos omni mente in ea cogitatione curaque versor ut odorer quam sagacissime possim quid sentiant quid existiment quid exspectent quid velint quo deduci oratione facillime posse videantur. 
4;Titolo4;Mario Gialli;Questo è il corpo dell'articolo quattro e sto scrivendo diverse cose di prova. Huic autem est illa dispar adiuncta ratio orationis quae alio quodam genere mentes iudicum permovet impellitque ut aut oderint aut diligant aut invideant aut salvum velint aut metuant aut sperent aut cupiant aut abhorreant aut laetentur aut maereant aut misereantur aut poenire velint aut ad eos motus deducantur si qui finitimi sunt et de propinquis ac talibus animi permotionibus. Atque illud optandum est oratori ut aliquam permotionem animorum sua sponte ipsi adferant ad causam iudices ad id quod utilitas oratoris feret accommodatam. Facilius est enim currentem ut aiunt incitare quam commovere languentem. Sin id aut non erit aut erit obscurius sicut medico diligenti priusquam conetur aegro adhibere medicinam non solum morbus eius cui mederi volet sed etiam consuetudo valentis et natura corporis cognoscenda est sic equidem cum adgredior in ancipiti causa et gravi ad animos iudicum pertractandos omni mente in ea cogitatione curaque versor ut odorer quam sagacissime possim quid sentiant quid existiment quid exspectent quid velint quo deduci oratione facillime posse videantur. 
5;Titolo5;Mario Tortora;Questo è il corpo dell'articolo cinque e sto scrivendo diverse cose di prova. Huic autem est illa dispar adiuncta ratio orationis quae alio quodam genere mentes iudicum permovet impellitque ut aut oderint aut diligant aut invideant aut salvum velint aut metuant aut sperent aut cupiant aut abhorreant aut laetentur aut maereant aut misereantur aut poenire velint aut ad eos motus deducantur si qui finitimi sunt et de propinquis ac talibus animi permotionibus. Atque illud optandum est oratori ut aliquam permotionem animorum sua sponte ipsi adferant ad causam iudices ad id quod utilitas oratoris feret accommodatam. Facilius est enim currentem ut aiunt incitare quam commovere languentem. Sin id aut non erit aut erit obscurius sicut medico diligenti priusquam conetur aegro adhibere medicinam non solum morbus eius cui mederi volet sed etiam consuetudo valentis et natura corporis cognoscenda est sic equidem cum adgredior in ancipiti causa et gravi ad animos iudicum pertractandos omni mente in ea cogitatione curaque versor ut odorer quam sagacissime possim quid sentiant quid existiment quid exspectent quid velint quo deduci oratione facillime posse videantur. 

news.php
<?php

header("Access-Control-Allow-Origin: *");
header("Content-Type: application/json; charset=UTF-8");

$handle = fopen("news.txt", "r");
$arrayNews = array();

for ($i = 0; $row = fgetcsv($handle ); ++$i) {
    $news = (explode(";", $row[0]));
	$arrayNews[$i]["indice"] = $news[0];
    $arrayNews[$i]["titolo"] = $news[1];
    $arrayNews[$i]["autore"] = $news[2];
    $arrayNews[$i]["body"] 	 = $news[3];
}

// Con questo le randomizzo dopo averle lette tutte
shuffle($arrayNews);

// chiudo la lettura del file
fclose($handle);


// Da qui intercetto la chiamata cURL ed in base al parametro restituisco ALL, ONE o di default ONE.
if ($_REQUEST['type'] == 'all'){
	echo json_encode($arrayNews);
}
else if ($_REQUEST['type'] == 'one'){
	echo json_encode($arrayNews[0]);
}
else
	echo json_encode($arrayNews[0]);

homenews.php
<html>
	<head>
    
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<title>Latest News</title>
		<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
		<link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700" rel="stylesheet">
		<link rel="stylesheet" type="text/css" href="css/stylenews.css">
		<link href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.0/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
		<script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.0/js/bootstrap.min.js"></script>
		<script src="//code.jquery.com/jquery-1.11.1.min.js"></script>
	</head>
	<body>
	<?php
	
	include("function.php");
	
	// Ottengo tutte le news contenute nel file 
	$get_all_news = callAPI(LOCALHOST.'news.php?type=all');
	$response_all_news = json_decode($get_all_news, true);
	
	// Ottengo solo una news
	$get_one_news = callAPI(LOCALHOST.'news.php?type=one');	
	$response_one_news = json_decode($get_one_news, true);
	
		
	?>
		<section class="home-blog bg-sand">
			<div class="container">
				<!-- section title -->
				<div class="row justify-content-md-center">
					<div class="col-xl-12 col-lg-13 col-md-13">
						<div class="section-title text-center title-ex1">
							<h2>Latest News</h2>
						</div>
					</div>
				</div>
				<div class="row ">
					<?php foreach($response_all_news as $news){ ?>
					
						<div class="col-md-6">
							<div class="media blog-media">						  
								<div class="media-body">
									<h5 class="mt-0"><?php echo $news['titolo'] ?></h5>
									<?php echo substr($news['body'],0,100) ?>
									<a href="<?php echo LOCALHOST."singolnews.php?id=".$news['indice'] ?>" class="post-link">Read More</a>
									<ul>
										<li>by: <?php echo $news['autore'] ?></li>
										
									</ul>
								</div>
							</div>
						</div>
					<?php } ?>
				</div><br>
				<div class="row ">
					<div class="col-md-10"></div>
					<div class="col-md-2"><a onClick="window.location.reload();" class="btn btn-info center-block">Load News</a></div>
				</div>
			</div>			
		</div>	
	</body>

singolnews.php
<html>
	<head>    
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<title>News</title>
		<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
		<link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700" rel="stylesheet">
		<link rel="stylesheet" type="text/css" href="css/stylenews.css">
		<link href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.0/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
		<script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.0/js/bootstrap.min.js"></script>
		<script src="//code.jquery.com/jquery-1.11.1.min.js"></script>

	</head>
	
	<body>
		<?php
	
		include("function.php");
		
		$id_news = $_GET['id'];
		
		// Ottengo tutte le news contenute nel file 
		$get_all_news = callAPI(LOCALHOST.'news.php?type=all');
		$response_all_news = json_decode($get_all_news, true);
		
		// init dell'array che conterrà la news trovata
		$singola_news = array();
		
		// Cerco la news per avente la chiave 'indice' passata in GET
		foreach($response_all_news as $news){
			if ($news['indice'] == $id_news){
				$singola_news = $news;
			}
		}
		
		?>
			<section class="home-blog bg-sand">
			
			<div class="container">
				<!-- section title -->
				<div class="row justify-content-md-center">
					<div class="col-xl-12 col-lg-13 col-md-13">
						<div class="section-title text-center title-ex1">
							<h2>News</h2>
						</div>
					</div>
				</div>
				<!-- section title ends -->
				<div class="row ">
					<div class="col-md-2"></div>
					<div class="col-md-8">
						<div class="media blog-media">						  
							<div class="media-body bw" >
								<a href=""><h5 class="mt-0 "><?php echo $singola_news['titolo'] ?></h5></a>
								<p><?php echo $singola_news['body'] ?></p>
								<br>
								<ul>
									<li>by: <?php echo $singola_news['autore'] ?></li>
									
								</ul>
							</div>
						</div>
					</div>
					<div class="col-md-2"></div>
				</div><br>
				<div class="row ">
					<div class="col-md-10"></div>
					<div class="col-md-2"><a onclick="history.go(-1)" class="btn btn-info center-block">Go Back</a></div>
				</div>
			</div>
		</div>
	</body>